codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/compression/PerMessageDeflateEncoder.java

```diff
@@ -63,7 +64,11 @@ protected boolean removeFrameTail(WebSocketFrame msg) {
     @Override
     protected void encode(ChannelHandlerContext ctx, WebSocketFrame msg,
             List<Object> out) throws Exception {
-        super.encode(ctx, msg, out);
+        if (msg instanceof UncompressedWebSocketFrame) {
+            out.add(((UncompressedWebSocketFrame) msg.retain()).toDataFrame());
```

Comment:
@sunng87 one nit... can we call the retain() on the returned frame just in case something goes wrong and `toDataFrame()` throws an exception ?

```java
out.add(((UncompressedWebSocketFrame) msg).toDataFrame().retain());
```